shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/cache/MetaDataCache.java

```diff
@@ -111,25 +116,22 @@ public void clean() {
      * @return the meta data
      */
     public MetaData obtain(final String path) {
-        final MetaData metaData = Optional.ofNullable(META_DATA_MAP.get(path))
+        final MetaData metaData = Optional.ofNullable(CACHE.get(path))
                 .orElseGet(() -> {
-                    final MetaData exist = CACHE.get(path);
-                    if (Objects.nonNull(exist)) {
-                        return exist;
-                    }
-                    final String key = META_DATA_MAP.keySet()
+                    final MetaData value = META_DATA_MAP.values()
```

Comment:
Adjusting the order of `META_DATA_MAP` and `CACHE` brings a problem: there is metadata in `META_DATA_MAP`, but it needs to be matched once before it can be put into `CACHE`.